daffodil-japi/src/main/scala/org/apache/daffodil/japi/Daffodil.scala

```diff
@@ -825,50 +524,15 @@ class DataProcessor private[japi] (private var dp: SDataProcessor)
     val ur = dp.unparse(input, output).asInstanceOf[SUnparseResult]
     new UnparseResult(ur)
   }
-
-  /**
-   * Unparse a JDOM2 infoset
-   *
-   * @param output the byte channel to write the data to
-   * @param infoset the infoset to unparse, as a jdom event cursor
-   * @return an object with contains the result and/or diagnostic information
-   */
-  @deprecated("Use unparse(InfosetInputter, WritableByteChannel)", "2.0.0")
-  def unparse(output: WritableByteChannel, infoset: org.jdom2.Document): UnparseResult = {
-    val input = new JDOMInfosetInputter(infoset)
-    unparse(input, output)
-  }
 }
 
 /**
- * Result of calling [[DataProcessor#parse(java.nio.channels.ReadableByteChannel, InfosetOutputter, long)]], containing
+ * Result of calling [[DataProcessor#parse(InputSourceDataInputStream, InfosetOutputter)]], containing
  * the diagnostic information, and the final data location
  */
 class ParseResult private[japi] (pr: SParseResult, deprecatedOutput: Maybe[JDOMInfosetOutputter])
   extends WithDiagnostics(pr) {
 
-  /**
```

Comment:
This `deprecatedOutput` variable can be removed. This was originally to maintain backwards compatibility with some of the deprecated `parse()` and below `result()` functions, but with those removed this can now be removed. Same in the Scala API.

Reply:
Should this go for any variable that has `deprecated` in its name?

Reply:
Is `separatorPolicy` also deprecated?

Reply:
Should all the items in `DeprecatedProperty` be removed as well

Reply:
`separatorPolicy` is a legacy property that is deprecated. Seems reasonable to remove support for all these deprecated properties as well, unless we want to do that separately. Maybe we only want to remove deprecated APIs for now since their use is likely pretty rare, and remove support for deprecated schema properties at some other point in the future since many schemas may still use these.

Reply:
That would make sense to me because a lot of the stuff I removed already wasn't being used by anything.

Reply:
@stevedlawrence Do you think anything other than `deprecatedOutput` and changing the default value of `allowExpressionResultCoercion` to false or do you think it is mostly schema stuff?

Reply:
I think it depends what our goal here is. If the goal is to ensure we don't break existing schemas and we're only removing deprecated APIs (which could require code changes but no schema changes), then we shouldn't change this tunable--changing it may require schema changes if schemas rely on this coercion.

Reply:
That makes sense to me. I think it would be best to do the schema changes in another issue/PR as I feel this one is quite large enough that splitting out the other changes would make sense.

Reply:
Or at least a separate change set, so they're more easily spotted and the change backed out if need be.